swarm/src/lib.rs

```diff
@@ -249,6 +249,16 @@ pub enum SwarmEvent<TBehaviourOutEvent, THandlerErr> {
     Dialing(PeerId),
 }
 
+impl<TBehaviourOutEvent, THandlerErr> SwarmEvent<TBehaviourOutEvent, THandlerErr> {
+    /// Extract the `TBehaviourOutEvent` from this [`SwarmEvent`] in case it is the `Behaviour` variant, otherwise fail.
+    pub fn try_into_behaviour_event(self) -> Result<TBehaviourOutEvent, Self> {
```

Comment:
```suggestion
    pub fn try_into_behaviour_event(self) -> Result<TBehaviourOutEvent, Self> {
```

Neat return signature!

Reply:
:pray: